frontend/src/components/HomePage/HomePage.tsx

```diff
@@ -1,9 +1,24 @@
 // @ts-ignore
-import ReactPannellum from "react-pannellum";
+import ReactPannellum, { isLoaded } from "react-pannellum";
 import panoramaImage from "./testimage.png";
+import Navbar from "../Navbar/Navbar";
+import LoadingScreen from "../LoadingScreen/LoadingScreen";
 import "./HomePage.scss";
+import { useEffect, useState } from "react";
 
 function HomePage() {
+  const [isPanoramaLoaded, setIsPanoramaLoaded] = useState(false);
+
+  useEffect(() => {
+    const checkLoaded = setInterval(() => {
+      if (isLoaded()) {
+        setIsPanoramaLoaded(true);
+        clearInterval(checkLoaded);
+      }
+    }, 2000);
+    return () => clearInterval(checkLoaded);
+  }, []);
```

Comment:
this polling-based approach isn't quite ideal - look into `onPanoramaLoaded` in [react-pannellum](https://github.com/hoaiduyit/react-pannellum)

tl;dr: pass a function (something like `() => setIsPanoramaLoaded(true)`) to the panorama itself via the `onPanoramaLoaded` prop. this function will run when the panorama has loaded. :^)